src/components/BookCard/index.js

```diff
@@ -24,54 +21,39 @@ import {
   UpdateProgress,
   BookLineDivider,
 } from './BookCardElements';
-import { removeBook } from '../../redux/books/books';
 
-const BookCard = ({ title, author }) => {
-  const dispatch = useDispatch();
-  const books = useSelector((state) => state.books);
-
-  const removeBookFromStore = (e) => {
-    const listItem = e.target.parentNode.parentNode;
-    const bookId = listItem.getAttribute('data-id');
-    const book = _.find(books, { id: bookId });
-
-    dispatch(removeBook(book));
-  };
-
-  return (
-    <BookContainer>
-      <BookCardWrapper>
-        <BookSectionOne>
-          <BookCategory>Prayer Book</BookCategory>
-          <BookTitle>{title}</BookTitle>
-          <BookAuthor>{author}</BookAuthor>
-          <BookCommandsWrapper>
-            <Comments>Comments</Comments>
-            <BookRemove onClick={removeBookFromStore}>Remove</BookRemove>
-            <BookEdit>Edit</BookEdit>
-          </BookCommandsWrapper>
-        </BookSectionOne>
-        <BookSectionTwo>
-          <CircleCenter />
-          <CompletedWrapper>
-            <CompletedPerc>8%</CompletedPerc>
-            <CompletedText>Completed</CompletedText>
-          </CompletedWrapper>
-        </BookSectionTwo>
-        <BookLineDivider />
-        <BookSectionThree>
-          <BookChapter>Current Chapter</BookChapter>
-          <BookLesson>Chapter3: ALessonLearned </BookLesson>
-          <UpdateProgress>Update progress</UpdateProgress>
-        </BookSectionThree>
-      </BookCardWrapper>
-    </BookContainer>
-  );
-};
+const BookCard = ({ book, removeBookFromStore }) => (
+  <BookContainer>
+    <BookCardWrapper>
+      <BookSectionOne data-id={book.id}>
+        <BookCategory>{book.category}</BookCategory>
+        <BookTitle>{book.title}</BookTitle>
+        <BookCommandsWrapper>
+          <Comments>Comments</Comments>
+          <BookRemove onClick={removeBookFromStore}>Remove</BookRemove>
+          <BookEdit>Edit</BookEdit>
+        </BookCommandsWrapper>
+      </BookSectionOne>
+      <BookSectionTwo>
+        <CircleCenter />
+        <CompletedWrapper>
+          <CompletedPerc>8%</CompletedPerc>
+          <CompletedText>Completed</CompletedText>
+        </CompletedWrapper>
+      </BookSectionTwo>
+      <BookLineDivider />
+      <BookSectionThree>
+        <BookChapter>Current Chapter</BookChapter>
+        <BookLesson>Chapter3: ALessonLearned </BookLesson>
+        <UpdateProgress>Update progress</UpdateProgress>
+      </BookSectionThree>
+    </BookCardWrapper>
+  </BookContainer>
+);
 
 BookCard.propTypes = {
-  title: PropTypes.node.isRequired,
-  author: PropTypes.node.isRequired,
+  book: PropTypes.node.isRequired,
+  removeBookFromStore: PropTypes.node.isRequired,
 };
 
 export default BookCard;
```

Comment:
- [ ] You want to consider fixing some console errors that appear because of the type of prop types. ♻️ 

![image](https://user-images.githubusercontent.com/79876959/145504937-74f02fc4-169b-4bf0-b8a1-20be53748cb4.png)

![image](https://user-images.githubusercontent.com/79876959/145505020-ba3bcbb4-0e3c-4b64-9bdd-91ade298188c.png)